### PR TITLE
Notifications - show Notifications Alert with basic conditions

### PR DESCRIPTION
**Pull request by marcinbodnar**
`const applicationServerKey = 'BCZN5uqYMBZ2VCV3y0F0emodyYRGyt5JTgfIIzYVXIHKSBwuG0kb0NpPA-DM4nfmfRFiFu-MpKS2eNG7bhxQWn0';`
Can be used for testing subscribing to push notifications.

Also, this version of the modal component needs to be used:
https://github.com/near/near-discovery-components/pull/324

### CODE REVIEW

**Comment by marcinbodnar** [comment]
@shelegdmitriy This is the modal component after small changes:
https://github.com/near/near-discovery-components/pull/324

**Review by shelegdmitriy — APPROVED** [review]

File: src/pages/index.tsx [hunk 1]
```diff
@@ -58,15 +72,35 @@ const HomePage: NextPageWithLayout = () => {
...
-        componentProps={{
-          logOut: authStore.logOut,
-          targetProps: componentProps,
-          targetComponent: components.default,
-          tosName: components.tosContent,
-        }}
-      />
+      <>
+        <VmComponent
+          src="near/widget/NearOrg.Notifications.NotificationAlert"
```
Comment (shelegdmitriy) [inline src/pages/index.tsx:77]: After #408 done we need to update this string:
```suggestion
          src={components.nearOrg.notifications.notificationAlert}
```

**Review by shelegdmitriy — CHANGES_REQUESTED** [review]

**Review by milenakrawczyk — APPROVED** [review]

File: src/pages/index.tsx [hunk 2]
```diff
@@ -58,15 +73,35 @@ const HomePage: NextPageWithLayout = () => {
...
+                setShowNotificationModalState(false);
+              }),
+            handleOnCancel: () => {
+              handleOnCancel();
+              setShowNotificationModalState(false);
+            },
+            isNotificationSupported,
+            isPermisionGranted,
+            isPushManagerSupported,
+            setNotificationsSessionStorage,
```
Comment (shelegdmitriy) [inline src/pages/index.tsx:92]: To close this modal by pressing `esc` add this line:
```suggestion
            setNotificationsSessionStorage,
            onOpenChange: () => setShowNotificationModalState(false),
```

**Review by shelegdmitriy — APPROVED** [review]
LGTM

**Pull request merged**